### PR TITLE
Gemfile

### PR DESCRIPTION
**Pull request by scottmacphersonmusic**
Paring the Gemfile down to only the necessary gems.

### CODE REVIEW

**Comment by scottmacphersonmusic** [comment]
@bion I'm in the process of cleaning up the Gemfile but I keep getting an error when I try to run the app in the browser.  It seems that it wants to use coffeescript, so I got rid of any `.coffee` extensions.  Do I need to change the asset configuration somehow?

**Comment by bion** [comment]
I'd have to look around a bit, but I think the general idea is to get rid
of or switch to .js anything that says coffee. There's probably some
discussion on SO.

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -13,4 +13,3 @@
 //= require jquery
 //= require jquery_ujs
 //= require bootstrap-sprockets
-//= require_tree .
```
Comment (scottmacphersonmusic) [inline app/assets/javascripts/application.js:16]: @Bion so by removing this (^^^) line I got the app to work again, but it seems like that undermines the asset pipeline.  I made sure to get rid of all the `.coffee` extensions i could find but the app seems to be trying to call coffee script from somewhere.  The error points to line 7 of `/app/views/layouts/application.html.haml` which reads:

`= javascript_include_tag 'application'`

**Comment by bion** [comment]
You probably need to get rid of the cached assets that Rails generates in development mode because they depend on coffeescript which is no longer there: `bundle exec rake assets:clobber`. Pay attention to the output to see what the command does (you can also run `bundle exec rake -T` to see all of the available rake tasks).

Also, you changed `bookmarks.coffee` to `bookmarks.js` without changing the comment syntax from `#` to `//`. `#` is not a valid comment in JS.

Don't forget to put the `require_tree .` back in either.

File: app/assets/stylesheets/application.css.sass [hunk 2]
```diff
@@ -1,5 +1,2 @@
-/* *= require_tree .
-/* *= require_self
```
Comment (bion) [inline app/assets/stylesheets/application.css.sass:2]: are you sure these are no longer needed?

Comment (scottmacphersonmusic) [inline app/assets/stylesheets/application.css.sass:2]: In the rails instructions for the`bootstrap-sass` [Readme](https://github.com/twbs/bootstrap-sass), it says to git rid of these (^^^) in favor of the `@import` syntax, but maybe I'd want them in there if I'm trying to use a custom stylesheet as well?

**Comment by scottmacphersonmusic** [comment]
@bion you were totally right - running `bundle exec rake assets:clobber` fixed it!  The output said that it ran this:

`rm -rf /Users/scottmacpherson/Projects/bookmarker/tmp/cache/assets`

I never would have figured that out and I'm sure glad to be aware that exists :)

**Comment by bion** [comment]
@scottmacphersonmusic it's a pretty steep learning curve but once you've got all the major components in your head this sort of thing starts to feel if not obvious at least mostly straightforward.

**Pull request merged**